plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java

```diff
@@ -2564,7 +2564,12 @@ public void dropView(ConnectorSession session, SchemaTableName viewName)
     @Override
     public List<SchemaTableName> listViews(ConnectorSession session, Optional<String> schemaName)
     {
-        return trinoViewHiveMetastore.listViews(schemaName);
+        return schemaName.map(Collections::singletonList)
+                .orElseGet(() -> listSchemaNames(session))
+                .stream()
+                .flatMap(schema -> metastore.getAllViews(schema).stream()
```

Comment:
What happens if the metastore is shared with Iceberg and the schema contains materialized views? 

ConnectorMetadata#listViews expects only Trino views, but HiveMetastore#getAllViews returns both views and materialized views.

Reply:
Changed implementation to avoid stumbling into this problem.